### DISCUSSION TITLE
Nested Optionals seem to be difficult to do

### DISCUSSION THREAD

**Discussion by jay-babu** [mason-org/mason.nvim#594] [Q&A] [answered]
Hello,

I needed to some nested Optionals as there could be multiple nil values. I was wondering if there was an easier way to accomplish the below. Still getting used to this library, but definitely makes my life easier.

```lua
local default_handler = Optional.of_nilable(handlers[1])
  :or_(_.always(Optional.of_nilable(default_setup(installed_sources))))
```

Thanks!

**Comment by williamboman** [comment, marked as answer]
Hey! So both the functional and optional modules are technically not part of the public Lua API, so use should be done with care (although I don't think it'll be an issue). Not sure what values your example involves, but if `default_setup()` never returns nil, I'd probably unwrap the optional immediately like so:

```lua
local default_handler = Optional.of_nilable(handlers[1]):or_else_get(function ()
    return default_setup(installed_sources)
end)

-- but you probably dont need to use the Optional interface
local default_handler = handlers[1] or default_setup(installed_sources)
```

**Reply by jay-babu** [reply]
Yeah,. I forgot to reply last night, but I realized I could just do 

`Optional.of_nilable(a or b or c)`

Feel free to resolve. Thanks!

**Reply by jay-babu** [reply]
@williamboman, I am okay with modules not being public / stable. It would nice if there was a way to subscribe to breaking changes, or even better to split that portion to another stable repo. I find it very useful 

Thanks!
